backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/ad/dao/impl/TypedSqlQuery.java

```diff
@@ -1231,7 +1228,7 @@ public final String buildSQL(
 						selectClause,
 						null/* don't assume the union-query's from-clause is identical! */,
 						null/* groupByClause */,
-						false/* useOrderByClause */);
+						true/* useOrderByClause */);
```

Comment:
what's the reason behind this?
Took a look at the test case (backend/de.metas.adempiere.adempiere/base/src/test/java/de/metas/dao/selection/QuerySelectionHelperTest.java) and i still have the feeling the change was not needed....

i.e. we have to sort the data only once, at the end after all UNIONs.

what am I missing?

Reply:
well, I needed it as I want each union query to be sorted by priority and limited to one result, also, if we refer to the general use case, I think it's great that we can allow each union query to be sorted by its own needs, plus, sorting the results only once is still possible by adding the `orderBy` caluse only to the "initial" query (i.e. the one to which we add the union queries).